packages/swap/src/ingress-egress-tracking/index.ts

```diff
@@ -99,29 +103,68 @@ export const getPendingBroadcast = async (broadcast: Broadcast) => {
   }
 };
 
-export const getPendingVaultSwap = async (network: ChainflipNetwork, txRef: string) => {
-  if (!redis) return null;
+export const getPendingVaultSwap = async (txRef: string) => {
+  const resultPromises = getTransactionRefChains(txRef).map(async (chain) => {
+    let result = redis ? await redis.getPendingVaultSwap(chain, txRef) : null;
+
+    if (!result && chain === 'Solana' && env.SOLANA_RPC_HTTP_URL) {
+      const txData = await findVaultSwapData(env.SOLANA_RPC_HTTP_URL, txRef);
+      result = txData && {
+        amount: BigInt(txData.depositAmount),
+        destinationAddress: txData.destinationAddress,
+        inputAsset: txData.sourceAsset as InternalAsset,
+        outputAsset: txData.destinationAsset as InternalAsset,
+        depositChainBlockHeight: null,
+        brokerFee: txData.cfParams.brokerFees,
+        affiliateFees: txData.cfParams.affiliateFees.map((fee) => ({
+          account: '', // transaction data includes registration id only
```

Comment:
i think we can live with this until we get an rpc on the node: https://linear.app/chainflip/issue/PRO-2095/add-rpc-to-decode-vault-swaps-payload